src/zope/testrunner/runner.py

```diff
@@ -451,11 +463,10 @@ def run_layer(options, layer_name, layer, tests, setup_layers,
         setup_layer(options, layer, setup_layers)
     except zope.testrunner.interfaces.EndRun:
         raise
+    except EXPLOSIVE_ERRORS:
+        raise
     except Exception:
```

Comment:
KeyboardInterrupt and SystemExit all inherit BaseException rather than Exception, and are not caught by the `except Exception` catch-all below.  I'm not sure how I feel about the redundant catch-and-release for them.

(I thought MemoryError was the same, but then I checked.)

Reply:
Ack - I suspect this code may have been very old, maybe even from before the BaseException rework.  Simplified.